src/CodeTool/ElasticSearch/DSL/Query/ElasticSearchDSLFetchSourceContext.php

```diff
@@ -57,6 +57,10 @@ public function excludes(string ...$excludes): self
         return $this;
     }
 
+    /**
+     * @return bool|array
+     */
+    #[\ReturnTypeWillChange]
     public function jsonSerialize()
```

Comment:
as an alternative, I can put here native union return type - `: bool|array`.

This won't break with PHP 9, but it can be considered as a BC break